src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -340,7 +342,20 @@ export class MockHttpSocket extends MockSocket {
 
     // Create a `ServerResponse` instance to delegate HTTP message parsing,
     // Transfer-Encoding, and other things to Node.js internals.
-    const serverResponse = new ServerResponse(new IncomingMessage(this))
+    const incomingMessage = new IncomingMessage(this)
+    
+    // Set the HTTP version on the IncomingMessage.
+    // This is critical for Node.js to correctly allow keep alive.
+    // By default, nodejs sets keepalive to false if the HttpVersion is not at least
+    // 1.1: https://github.com/nodejs/node/blob/70f6b58ac655234435a99d72b857dd7b316d34bf/lib/_http_server.js#L211C1-L211C34
+    // The version should normally already have been set by the onRequestStart method
+    if(this.httpVersionMajor != null && this.httpVersionMinor != null) {
```

Comment:
Hm, this is interesting! From what I gathered, this `IncomingMessage` was purely internal to help us handle the mock response and shouldn't influence the user-facing behaviors in any way. So you say it does?

This might be an indicator that we should consider migrating from this.

All `ServerResponse` does here is help us parse out the response. User-facing `ClientRequest` constructs its own instances, including `IncomingMessage`, and ours never leaks or affects the user-facing data in any way. At least, that is the intention.